### ISSUE TITLE
NVDA incorrectly re-labels `div` containing hidden children with accessibility `aria-label`

### ISSUE THREAD

**Issue by tgeczy** [nvaccess/nvda#15847]
### Steps to reproduce:
1. Create a test document (provided) whereby all items in the `div` are `span` elements that use `aria-hidden` set to true. Note that this is incorrect markup: a presentation role is more suitable in this scenario, but we know the web is not made all too well, either, and it's not a screen reader's job to make failures obvious to end-users. 
2. Assign the `div` a new label. This can be the same or different text as to what is hidden, this should not matter.
3. Test whether Browse mode causes the element to read with the label after opening the page in Chrome, Firefox, and Edge.
### Actual behavior:

NVDA announces: "testing this as a new label to the element" as the label, or "testing this label" if the `aria-label` is set to the same text as the one getting hidden. 
### Expected behavior:

In either case, NVDA incorrectly applies the label in a situation when it should not. Only the page title should be read, with no on-page contents present. The `div` should map to the `generic` role, and non-interactive elements cannot be assigned an `aria-label` per the WAI-ARIA spec. If you remove all of the hidden `span` children from any of these test documents, NVDA will behave correctly and not read an accessible label at all on the div, even if the label is kept in the document.
### NVDA logs, crash dumps and other attachments:
[relabeling_test.zip](https://github.com/nvaccess/nvda/files/13483166/relabeling_test.zip)
relabeling_test.zip (1254 bytes,) SHA1: 5905D079237A1091D677BF75B5CFFFA94D604860
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
NVDA 2023.3
#### Windows version:
Windows 11 (23H2 update, build 22631)
#### Name and version of other software in use when reproducing the issue:
Firefox 120, Chrome 119. 
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
2023.1

#### If NVDA add-ons are disabled, is your problem still occurring?
yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
yes

**Comment by seanbudd** [comment]
This is a web authoring mistake, closing as won't fix.
